Uploaded script for Uniprot to ncbi function addresses issue #109

SunSummoner opened the pull request:
<!-- _modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_
_referenced with modifications from [pycytominer](https://github.com/cytomining/pycytominer/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_ -->

# Description

<!--
Thank you so much for your contribution to <NewPackage>!

Please _succinctly_ summarize your proposed change.
Namely, consider addressing the following questions:

- What motivated you to open this pull request?
- Were there any special adjustments you had to make to complete the work?
- Are there any issues which are related to this pull request (you may used a `#<digit>` to reference GitHub issues as links within this description)?

Also, if you haven't already, please use `pre-commit run --all-files` to help check your files using this project's pre-commit configuration.
Pre-commit checks will automatically run as part of opening this pull request and we seek to ensure all checks pass before merging changes.
-->

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [ ] I have added comments to my code to help provide understanding.
- [ ] I have added a test which covers the code changes found within this PR.
- [ ] I have deleted all non-relevant text in this pull request template.
- [ ] **Reviewer assignment**: Tag a relevant team member to review and approve the changes.

SunSummoner committed:
Uploaded script for Uniprot to ncbi function.

SunSummoner committed:
Implement mapping functions for protein IDs

Added functions to map NCBI RefSeq Protein IDs to UniProt Accessions and NCBI Protein Accessions to IPG IDs.

the-mayer reviewed (changes requested):
Great start! Everything works as described!

A couple of suggestions before approving/merging:
- Let's avoid using `library()` and `requireNamespace()` in user facing functions. Instead, we should rely on roxygen import tags to handle package dependencies.
- MolEvolvR already has quite a few dependencies, it might be worth refactoring `acc2ipg()` to use `httr2`. That would let us drop `httr` and simplify the dependencies for the package

Thanks for putting this together and apologies for the delay in getting the review back to you!

SunSummoner committed:
Used @importFrom and chnaged to consistent convention

Renamed functions and updated documentation for consistency. Used @importFrom for sourcing and changed function and variable names to match the convention. Removed other inconsistencies

SunSummoner committed:
Remove example usage from acc2Ipg documentation

Removed commented-out example usage from documentation.

SunSummoner committed:
Merge branch 'JRaviLab:main' into main

jananiravi commented on R/mapping_functions.R:
```diff
@@ -0,0 +1,233 @@
+#' @param uniprot_ids A character vector of UniProt accession IDs.
+#' @return A tibble with UniProt IDs and mapped NCBI accessions.
+#' @examples
+#' up2ncbi_fast(c("P12345", "Q9Y263"))
```
@SunSummoner has this been taken care of, too?

jananiravi commented:
have these been tested on example accessions -- assuming local checks are now working? If so, I'd be happy to merge this right away. thank you, @SunSummoner!

SunSummoner commented:
@jananiravi the example accessions are working! Thank you!
